File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/filter/primitive/DoubleFilterParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.primitive;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.DoubleFilter;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.FilterParser;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.finder.Finder;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.Validator;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.impl.DoubleValidator;

@Component
public class DoubleFilterParser implements FilterParser<DoubleFilter, Double> {

    @Override
    public Validator<Double> createChecker(DoubleFilter annotation, Finder finder) {
        return new DoubleValidator(annotation.lessThan(),
                                   annotation.lessOrEqual(),
                                   annotation.moreThan(),
                                   annotation.moreOrEqual(),
                                   annotation.equal());
    }

    @Override
    public Class<DoubleFilter> getAnnotationClass() {
        return DoubleFilter.class;
    }

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/core/controller/annotations/parser/filter/primitive/DoubleFilterParser.kt
package io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.filter.primitive

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.DoubleFilter
import io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.filter.FilterParser
import io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.finder.Finder
import io.github.oleksivio.tl.kbot.core.core.controller.handler.check.Validator
import io.github.oleksivio.tl.kbot.core.core.controller.handler.check.impl.DoubleValidator
 

class DoubleFilterParser :
    FilterParser<DoubleFilter, Double> {

    override val parserAnnotationClass = DoubleFilter::class

    override fun createChecker(annotation: DoubleFilter, finder: Finder): Validator<Double> {
        return DoubleValidator(
            annotation.lessThan,
            annotation.lessOrEqual,
            annotation.greatThan,
            annotation.greatOrEqual,
            annotation.equal
        )
    }

}
